FtpClientApp/FtpClientApp/FileUpload.cs

```diff
@@ -1,10 +1,83 @@
+        public String getFileName()
+        {
+            String filepath = Console.ReadLine();
+            return filepath;
+        }
+
+        //setup() function takes 2 strings as input: file to be uploaded and location on server 
+        //returns a String which will be 'success' in the case File is successfully uploaded 
+        //Otherwise the string will contain a relevant error message.
+        public String setup(String filetobeuploaded, String locationonserver)
+        {
+            try
+            {
+                if (!File.Exists(filetobeuploaded))
+                {
+                    Console.WriteLine("File does not exist. Please enter valid file path");
+                    return "disconnect";
+                }
+                String input = filetobeuploaded;
+                String pattern = @"/";
+                String[] elements = System.Text.RegularExpressions.Regex.Split(input, pattern);
+                string lastItem = elements[elements.Length - 1];
+                Console.WriteLine(lastItem);
+
+                String serverpath = locationonserver + "/" + lastItem;
+
+                String extension = Path.GetExtension(lastItem);
+                //Console.WriteLine("GetExtension('{0}') returns '{1}'", lastItem, extension);
+                if ((extension != ".txt") && (extension != ".jpg") && (extension != ".png"))
+                {
+                    Console.WriteLine("Please enter one of the following file formats only :.txt, .jpg, .png");
+                    return "disconnect";
+                }
+
+                WebClient request1 = new WebClient();
+                request1.Credentials = new NetworkCredential(this.connection.UserName, this.connection.PassWord);
+                byte[] responseArray = request1.UploadFile(serverpath, filetobeuploaded);
+                //this.wrapper.setRequest(request1);
+                Console.WriteLine(lastItem);
+            }
+            catch (WebException e)
+            {
+                if (e.Message.ToString().Equals("The remote server returned an error: (550) File unavailable (e.g., file not found, no access)."))
+                {
+                    Console.WriteLine("The server sent an error code of 550 \n The directory may not exist on the Server or \n Please check local file path and provide in Drive:xyz / abc.txt path \n");
+                    return "The server sent an error code of 550. The directory may not exist on the Server";
+                }
+                /*else
+                {
+                    Console.WriteLine("The server sent an error code of 550 \n The directory may not exist on the Server or \n Please check local file path and provide in Drive:xyz / abc.txt format \n ");
+                    return "Please check local file path and provide in Drive:xyz/abc.txt path ";
+                }*/
+                return e.Message.ToString();
```

Comment:
I would print this out. If there is an error it is good to indicate what is going on. For example I could not get it to upload a file and I was not getting anything back that indicated what was going on. Check out my picture in slack for more details. Essentially, it was saying "Could not Upload file due to above error" but there was no error above. Just the console.Writeline after you split the local file path. Slack me if any of this doesn't make sense.

Reply:
Uncomment the Else block, but instead of printing 
`Console.WriteLine("The server sent ...")` 
do something more like this:
`else
{
       Console.WriteLine(e.Message.ToString());
}`
This gives the user more detailed feedback what is going wrong. Currently, the only error message that will be printed will be "The server sent and error code of 550." If the error is anything other than that one error, then it won't print at all.

Reply:
Fixed